### PR TITLE
feat: add support for VARCHAR columns

### LINKED ISSUE #210: Unsupported Nullable Data Type VARCHAR(500) when Reading From Hyper
**Describe the bug**
Pantab is not able to read a hyperfile into a dataframe when the Datatype is a VarChar(500)-NULLABLE.
Error:
`TypeError: Column "First Name" has unsupported datatype VARCHAR(500) with nullability Nullability.NULLABLE`

**To Reproduce**

Create the hyper file: 
```python
import os
import pantab
from tableauhyperapi import HyperProcess, Telemetry, Connection, CreateMode, \
    NOT_NULLABLE, NULLABLE, SqlType, TableDefinition, Inserter, escape_string_literal, escape_name, \
    HyperException, TableName

HYPER_FILE = '/Users/username/Documents/test.hyper'

hubspot_contacts = TableDefinition(
    table_name="Contacts",  # Table name within the hyper file
    columns=[ 
        # contacts table
        TableDefinition.Column("Contact ID", SqlType.big_int(), NULLABLE),
        TableDefinition.Column("Company ID", SqlType.big_int(), NULLABLE),
        TableDefinition.Column("First Name", SqlType.varchar(500), NULLABLE),
        TableDefinition.Column("Last Name", SqlType.varchar(500), NULLABLE),
        TableDefinition.Column("Email", SqlType.varchar(500), NULLABLE),
        TableDefinition.Column("Job Title", SqlType.varchar(500), NULLABLE)
    ]
)

with HyperProcess(telemetry=Telemetry.DO_NOT_SEND_USAGE_DATA_TO_TABLEAU, parameters={'log_config': ''}) as hyper:
    with Connection(hyper.endpoint, HYPER_FILE, CreateMode.CREATE_AND_REPLACE) as connection:
        connection.catalog.create_table(hubspot_contacts)

        with Inserter(connection, hubspot_contacts) as inserter:
            inserter.add_row([1, 1, "John", "Doe", "test@test.com", "CEO"])
            inserter.execute()
```

Use pantab to read the file:\
`hs_contacts = pantab.frame_from_hyper(HYPER_FILE, table=TableName('public', 'Contacts'))`

**Expected behavior**
I should be able to turn this hyperfile into a pandas dataframe using the "frame_from_hyper" method. Using the tableauHyperApi directly, I can read the data from the file. 

**Desktop (please complete the following information):**
 - OS: Sonoma 14.1.2 (23B92)
 - pantab 3.0.1  pypi_0 pypi
 - tableauhyperapi  0.0.17971 pypi_0  pypi
 - Conda env built with ARM-64 Rosetta for compatibility with m1 chip

### PR DESCRIPTION
**Pull request by jorwoods**
closes #210

### CODE REVIEW

File: pantab/_types.py [hunk 1]
```diff
@@ -62,3 +62,17 @@
...
     _ColumnType(tab_api.SqlType.text(), tab_api.Nullability.NOT_NULLABLE)
 ] = "string"
+
+
+def _get_pandas_type(column_type: _ColumnType) -> str:
+    if column_type in _pandas_types:
+        return _pandas_types[column_type]
+
+    varchar = tab_api.SqlType.varchar(65535)
+    if column_type.type_.tag == varchar.tag:
```
Comment (WillAyd) [inline pantab/_types.py:72]: I think you can just simplify this as `if column_type.type_.tag == tab_api.TypeTag.VARCHAR`

https://tableau.github.io/hyper-db/lang_docs/py/tableauhyperapi.html#tableauhyperapi.TypeTag

File: pantab/tests/test_types.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import tableauhyperapi as tab_api
+
+import pantab._types
+
+
+def test_read_varchar_type():
```
Comment (WillAyd) [inline pantab/tests/test_types.py:6]: Can you just parametrize this test instead?

```python
@pytest.mark.parametrize("nullable", [tab_api.Nullability.NULLABLE, tab_api.Nullability.NOT_NULLABLE])
def test_read_varchar_type(nullable):
   ...
```

should do the trick

**Comment by WillAyd** [comment]
Would be helpful if you could also add a changelog note. You can do this either directly in the docs or add a newfragment for towncrier

File: pantab/_types.py (hunk 1 above)
Comment (jorwoods) [reply pantab/_types.py]: Simplified the check.

File: pantab/tests/test_types.py (hunk 2 above)
Comment (jorwoods) [reply pantab/tests/test_types.py]: Changed the tet to be parameterized.

File: doc/source/changelog.rst [hunk 3]
```diff
@@ -1,6 +1,11 @@
 Changelog
 ^^^^^^^^^
 
+Pantab 3.0.3 (2023-12-18)
```
Comment (WillAyd) [inline doc/source/changelog.rst:4]: Minor issue with putting a date - readthedocs picks up changes to our doc suite here and publishes automatically. So if you put a date of 12-18 on this people may think version 3.0.3 has already been released if they look at the docs.

So for now let's just put this as date of TBD; I'm also not sure if this warrants a bugfix release (3.0.3) or a minor release (3.1.0) - just depends on if you consider this a bug or a new feature

Comment (jorwoods) [reply doc/source/changelog.rst:4]: Ok, I’ll remove the date shortly. Regarding the minor/patch, do you consider this more of adding new functionality, or filling in something that was missing?

Comment (WillAyd) [reply doc/source/changelog.rst:4]: Did you want to push this to pypi today? I guess with our new build process there isn't that much downside to just releasing a 3.0.3 now

**Review by WillAyd — APPROVED** [review]

**Pull request merged**